### ISSUE TITLE
[Feature Request] Add a sound (maybe even user selectable .wav) when a task is checked complete

### ISSUE THREAD

**Issue by SLYtiger16** [nextcloud/tasks#991]
### Steps to reproduce
1. Click to mark a task or sub-task as complete.

### Expected behaviour
User hears a chime, ding, hoo-rah, beep, bell, etc.

### Actual behaviour
Sad and unstimulating silence.....

**Comment by raimund-schluessler** [comment]
I must say, I am a bit skeptical. I am not aware that Nextcloud uses sounds anywhere (could be because I have sound turned off mostly 🙈), and I would like to stay consistent with the rest of Nextcloud here. If there was a common Nextcloud standard for playing sounds, I would be open to use it here.
@jancborchardt Is this something we have / want?

What we will definitely not do is make it user selectable. We go for sensible default values and don't add options for every detail. Also, it would be way to much implementation effort for such a small detail.

**Comment by SLYtiger16** [comment]
Its a personal want. I have migrated from Microsoft To-Do, which has a
chime on task completion and is extremely satisfying for someone like me
who operates off dozens of tasks per day.

**Comment by hundsboog** [comment]
Definitely is this something a lot of users would have and/or are used to from other apps. At least there should be an option to select/mute/customize a sound.

Im sure, many people are coming from the dying "Wunderlist" and got used to a certain workflow, which made "Wunderlist" this sucessful.

**Comment by jancborchardt** [comment]
We currently do not use any sounds except in Talk. I’m not fully against it, but it can’t be too tacky or loud.

As a step towards it, what we can do:
- Some nice short animation on checking off a task, similar to e.g. how Twitter does the small animation for favoriting a post
- Having a nicer congratulatory emptycontent view of sorts if you are done with all the tasks for today :)

**Comment by SLYtiger16** [comment]
I can tell you that I am rolling nextcloud out to 400 people at my employer
in the next 3 weeks and most of the to-do list users are happily using
Microsoft To-Do.

Staff noted competitive advantages M. To Do has over NC Tasks are: 1.  A
subtle chime on checking a task. And 2.  A zero configuration mobile app
with push notification to shares. That's it. That's all they could come up
with. :)

Tasks has many features over other to-do list applications but a chime has
been mentioned many times to me by staff and I agree. Something subtle and
very short but just enough to add the psychological endorphins boost when
it's heard around the office and personally. Amazing what a small sounds
does for people's mindset.

**Comment by raimund-schluessler** [comment]
> As a step towards it, what we can do:
> 
>   * Some nice short animation on checking off a task, similar to e.g. how Twitter does the small animation for favoriting a post
> 
>   * Having a nicer congratulatory emptycontent view of sorts if you are done with all the tasks for today :)

Makes sense to do this. Could you just post a video of the animation you think of here? I don't use Twitter.

> Something subtle and very short but just enough to add the psychological endorphins boost when it's heard around the office and personally. Amazing what a small sounds does for people's mindset.

We can probably add a sound (I don't have any experience here, but it shouldn't be to hard), but the question is where to get a free sound to play? I am not a sound designer 😉

**Comment by jancborchardt** [comment]
This is the animation of Twitter’s like:
![Twitter love](https://user-images.githubusercontent.com/925062/82700971-8a129d00-9c6f-11ea-9c17-3d54d62633a1.gif)

As for sounds, for Talk we use the [Librem 5 sounds by feandesign](https://soundcloud.com/feandesign/sets/librem-5-sounds). None of these sounds seem fitting for this though as they are all a bit too long. Maaaybe just maybe the "Librem Text Message".

Another possibly fitting sound I found is this simple "ping" sound which is under CC0: https://freesound.org/people/edsward/sounds/341871/

@SLYtiger16 @hundsboog do you have a sample of the "task done" sound of Wunderlist and Microsoft To Do?

**Comment by raimund-schluessler** [comment]
> This is the animation of Twitter’s like:
> ![Twitter love](https://user-images.githubusercontent.com/925062/82700971-8a129d00-9c6f-11ea-9c17-3d54d62633a1.gif)

This looks quite complex. I guess this is not only CSS, isn't it?

> As for sounds, for Talk we use the Librem 5 sounds by feandesign. None of these sounds seem fitting for this though as they are all a bit too long. Maaaybe just maybe the "Librem Text Message".

This sounds friendly, but a bit unusual for checking a task, I think.

> Another possibly fitting sound I found is this simple "ping" sound which is under CC0:

This sounds more like checking a task, but I think it is bit metallic and hard. Sounddesign sure is fun 🙈 

If we really want to play a sound, we will definitly need a setting, since not everyone wants to have sounds. As a first implementation we could just add a checkbox in the Tasks setting area.

For later I wonder whether it might make sense to have a sounds section in the general settings of Nextcloud server. Other apps might also want to play a sound when some trigger occurs. E.g. the Notifications app could play a sound for a new notification, Calendar could play something when an event is happening, Files when you delete a file, Mail for new mails (just quick examples). An app would register a sound to play which then shows up in the settings section under the apps name. You could then upload a different sound file, disable playing a sound for this action or disable all sounds.

This could for sure be a standalone app, which you enable to have sounds played in apps which support it. Would that make sense, @nextcloud/designers?

**Comment by hundsboog** [comment]
@jancborchardt
I looked everywhere but couldn't find the exact sound. It was something like a high bell chime and this one comes close to what Wunderlist had:
https://www.youtube.com/watch?v=v-aqBz6C4D4
Pretty satisfying, while checking a task :-)
Btw, I opened up the Microsoft App on Android, there ist no sound...

@raimund-schluessler
I think a customizable sound section would be a great idea. There are users who get motivated by sound and others don't, so +1 for this feature.

**Comment by SLYtiger16** [comment]
> I looked everywhere but couldn't find the exact sound. It was something like a high bell chime and this one comes close to what Wunderlist had:
https://www.youtube.com/watch?v=v-aqBz6C4D4
Pretty satisfying, while checking a task :-)
Btw, I opened up the Microsoft App on Android, there ist no sound..

This is pretty good, satisfying for sure. Short and sweet. 

@jancborchardt - My android MSFT app plays a sound, weird, so does the browser.

**Comment by jancborchardt** [comment]
> I looked everywhere but couldn't find the exact sound. It was something like a high bell chime and this one comes close to what Wunderlist had:
> https://www.youtube.com/watch?v=v-aqBz6C4D4

Very similar sound, just a bit higher: https://freesound.org/people/InspectorJ/sounds/411089/

> This could for sure be a standalone app, which you enable to have sounds played in apps which support it. Would that make sense, @nextcloud/designers?

No, it’s not an app, it’s just a fringe functionality. :) For now a setting in the apps which use this makes sense – which for now is only Tasks. Then we can think about standardizing it for core and making it a setting in there. But it’s not a standalone app.

**Comment by alderson59** [comment]
Here's the chime you're looking for! Two files in the zip. The M4A is the original, and the WAV's converted.

[Wunderlist-chime.zip](https://github.com/nextcloud/tasks/files/5286547/Wunderlist-chime.zip)

**Comment by hundsboog** [comment]
> Here's the chime you're looking for! Two files in the zip. The M4A is the original, and the WAV's converted.
> 
> [Wunderlist-chime.zip](https://github.com/nextcloud/tasks/files/5286547/Wunderlist-chime.zip)

Thanks!!!!!! THIS is it! Can we please please have it in the Tasks App? I suspect a extreme performance boost for all users!!
